### ISSUE TITLE
Boolean symbol check with interesting side effects

### ISSUE THREAD

**Issue by tresni** [bbatsov/rubocop#4830]
rubocop the following script
```
#! /usr/bin/env ruby

ICONS = {
  true: 'This is true',
  false: 'This is false'
}.freeze
```

--------

## Expected behavior

No warnings or errors

## Actual behavior

Check added in #4515 returns a very weird error when Boolean values are used as hash keys.

```
» rubocop test.rb                              « [master|✚2…] (2.0.0-p648)
/usr/local/opt/rbenv/versions/2.0.0-p648/lib/ruby/gems/2.0.0/gems/ruby-progressbar-1.9.0/lib/ruby-progressbar/refinements/enumerator.rb:4: warning: Refinements are experimental, and the behavior may change in future versions of Ruby!
Inspecting 1 file
W

Offenses:

test.rb:4:3: W: Symbol with a boolean name - you probably meant to use rue.
  true: 'This is true',
  ^^^^
test.rb:5:3: W: Symbol with a boolean name - you probably meant to use alse.
  false: 'This is false'
  ^^^^^

1 file inspected, 2 offenses detected
```

## Steps to reproduce the problem

Lint the codeblock above.

## RuboCop version

```
$ rubocop -V
/usr/local/opt/rbenv/versions/2.0.0-p648/lib/ruby/gems/2.0.0/gems/ruby-progressbar-1.9.0/lib/ruby-progressbar/refinements/enumerator.rb:4: warning: Refinements are experimental, and the behavior may change in future versions of Ruby!
0.50.0 (using Parser 2.4.0.0, running on ruby 2.0.0 x86_64-darwin16.6.0)
```

**Comment by jonas054** [comment]
You're not using Boolean keys. The code is equivalent to
```ruby
ICONS = {
  :true => 'This is true',
  :false => 'This is false'
}.freeze
```
So the keys are symbols, but you thought you were using Booleans. I'd say the cop was right to warn you.

**Comment by tresni** [comment]
even if the cop is correct, the suggestion would not be helpful..  Replacing the keys with `rue` and `alse` would not result in the expected behavior either.

**Comment by akhramov** [comment]
The problem is in the warning message. Will fix it.

**Comment by ianfixes** [comment]
Should I take this to mean that a hash is _never_ allowed to have symbol keys called `true:` and `false:`?

**Comment by jonas054** [comment]
@ianfixes Yes, according to the `Lint/BooleanSymbol` cop it's risky and possibly a mistake to use the symbols `:true`or `:false`. https://docs.rubocop.org/en/stable/cops_lint/#lintbooleansymbol

**Comment by ianfixes** [comment]
Unlike the use case in the description, I have a legitimate need to map the _symbol_ `:true` (which I receive from a 3rd-party library) to some value.  

So I'm not so much asking what the docs say as trying to put a fine point on whether a hash key counts as a use case that wasn't expected in the development of this cop.  

Should there be a configurable option for allowing hash keys?  The alternative for me is to just disable this cop on a few files.

**Comment by jonas054** [comment]
The original PR #4515 doesn't mention hash keys, so I can't say for sure whether it's an expected case. And while it's true that the use of symbols as hash keys is very common, and there's a higher risk for false positives for those, I still think that the special case of hash keys `:true` and `:false`is too rare to warrant extra logic or configuration options in the cop.

So go with the disabling @ianfixes. That's my suggestion.
